vignettes/communicate.Rmd

```diff
@@ -0,0 +1,349 @@
+#' @export
+number_add <- function(x, y) {
+  x + y
+}
+```
+
+If you are renaming many functions as part of an API overhaul, it'll often make sense to document all the changes in one file, like <https://rvest.tidyverse.org/reference/rename.html>.
+
+### Supersede a function
+
+Superseding a function is simpler than deprecating it, since you don't need to steer users away from it with a warning.
+So all you need to do is add a superseded badge:
+
+```{r}
+#' Gather columns into key-value pairs
+#'
+#' @description
+#' `r lifecycle::badge("superseded")`
+```
+
+Then describe why the function was superseded, and what the recommended alternative is:
+
+```{r}
+#'
+#' Development on `gather()` is complete, and for new code we recommend
+#' switching to `pivot_longer()`, which is easier to use, more featureful,
+#' and still under active development.
+#' 
+#' In brief,
+#' `df %>% gather("key", "value", x, y, z)` is equivalent to
+#' `df %>% pivot_longer(c(x, y, z), names_to = "key", values_to = "value")`.
+#' See more details in `vignette("pivot")`.
+```
+
+The rest of the documentation can stay the same.
+
+### Mark function as experimental
+
+To advertise that a function is experimental and the interface might change with deprecation, first add an experimental badge to the description:
+
+```{r}
+#' @description
+#' `r lifecycle::badge("superseded")`
+```
+
+We don't currently provide automated signals for experimental functions because it's not clear how they would work, or if the effort to use them consistently would be worth the payoff.
+See discussion in <https://github.com/r-lib/lifecycle/issues/44>.
+
+## Arguments
+
+### Deprecate an argument, keeping the existing default
+
+Take this example where we want to deprecate `na.rm` in favour of always making it `TRUE.`
+
+```{r}
+add_two <- function(x, y, na.rm = TRUE) {
+  sum(x, y, na.rm = na.rm)
+}
+```
+
+First, add a badge to the argument description:
+
+```{r}
+#' @param na.rm `r lifecycle::badge("deprecated")` `na.rm = FALSE` is no
+#'   longer supported; this function will always remove missing values
+```
+
+And add a deprecation warning if `na.rm` is FALSE.
+In this case, there's no replacement to the behaviour, so we instead use `details` to provide a custom message:
+
+```{r}
+add_two <- function(x, y, na.rm = TRUE) {
+  if (!isTRUE(na.rm)) {
+    lifecycle::deprecate_warn(
+      when = "1.0.0", 
+      what = add_two(na.rm),
+      details = "Ability to remove missing values will be dropped in next release."
+    )
+  }
+  
+  sum(x, y, na.rm = na.rm)
+}
+
+add_two(1, NA, na.rm = TRUE)
+add_two(1, NA, na.rm = FALSE)
+```
+
+### Deprecating an argument, providing a new default
+
+Alternatively, you can change the default value to `lifecycle::deprecated()` to make the deprecation status more obvious from the outside, and use `lifecycle::is_present()`[^3] to test whether or not the argument was provided.
+
+[^3]: Unlike `missing()`, this works for both direct and indirect calls.
```

Comment:
I think this is important and short enough to put in the main text, i.e. no footnote.